code/game/gamemodes/management/event/combat.dm

```diff
@@ -11,8 +11,21 @@
 
 /datum/game_mode/combat/post_setup()
 	..()
+	//No more OOC
+	GLOB.ooc_allowed = FALSE
+	to_chat(world, "<B>Due to gamemode, the OOC channel has been globally disabled.</B>")
+	to_chat(world, "<span class='userdanger'>Round will end in 40 minutes.</span>")
+
+	//Breach all
 	for(var/mob/living/simple_animal/hostile/abnormality/A in GLOB.mob_list)
 		INVOKE_ASYNC(A, /mob/living/simple_animal/hostile/abnormality.proc/BreachEffect)
 		var/obj/effect/proc_holder/spell/targeted/night_vision/bloodspell = new
 		A.AddSpell(bloodspell)
 		A.faction += "hostile"
+	if(SSmaptype.maptype in SSmaptype.autoend)
+		addtimer(CALLBACK(src, .proc/loseround), 40 MINUTES)
+
+/datum/game_mode/combat/proc/loseround()
+	SSticker.force_ending = 1
+	for(var/mob/M in GLOB.player_list)
+		to_chat(M, "<span class='userdanger'>Players have taken too long! Round automatically ending.</span>")
```

Comment:
Why does this loop through mobs instead of using `to_chat(world, ...)` like the other ones? Also these could probably be slightly flavored. I'll make suggestions for them. If you don't like them give me a ping and we'll get this approved.